### DISCUSSION TITLE
tabella decodifica 39  - scelta cognome

### DISCUSSION THREAD

**Discussion by annamaraspin** [italia/ansc#731] [Supporto servizi cooperativi] [answered]
Buonasera,
non riesco a capire se i dati evidenziati sono un doppione oppure se hanno un significato diverso. Nel secondo caso vi chiederei anche la differenza perché mi sfugge

![image](https://github.com/italia/ansc/assets/127182190/6aad8daf-366a-4389-a2ed-bae6bb3aa76e)

Inoltre, nei casi TRASCR_002 e TRASCR_025 nella webapp nella lista valori vedo solo i primi 3 record della tabella 39. Giusto che tutti gli altri record non siano visibili/selezionabili? Nei mapping non ho trovato indicazioni di filtri.

Grazie

**Comment by amusarra** [comment, marked as answer]
Salve @annamaraspin 
Cero di rispondere ai dubbi subito a seguire.

La decodifica è suddivisa in questi sottoinsiemi:
Primo:
- id="1" value="mantenere cognome precedente" 
- id="2" value="aggiungerlo al cognome"
- id="3" value="anteporlo al cognome"
- id="4" value="sostituire il cognome precedente"

Secondo:
- id="6" value="Modifica cognome"
- id="7" value="Modifica nome" 
- id="8" value="Modifica nome e cognome"
- id="9" value="Aggiunta al cognome" 
- id="10" value="Aggiunta al nome" 

Terzo:
- id="11" value="padre sostituendolo a quello della madre" 
- id="12" value="padre aggiungendolo a quello della madre"

Semanticamente la decodifica **2 = aggiungere al cognome** e **id="9" value="Aggiunta al cognome"** non hanno differenza.

_La differenza però sta nella configurazione dei casi d'uso_ in quanto questa decodifica 39, non si presenta mai completa ma sempre filtrata.

In definitiva NON sono equivalenti, poiché un caso d'uso può essere configurato con il valore 2 per aggiungere il cognome un altro con il valore 9.
 
Per `evento.datiEventoRiconoscimento.tipoSceltaCognome` il sistema si aspetta il primo sottoinsieme (1,2,3,4) condizionando le operazioni da eseguire a valle di una annotazione nonché il testo. (form. 146).
 
Per `evento.datiAnnotazioneModificativa.tipoModifica` il sistema ANSC si aspetta il secondo sottoinsieme (6,7,8,9,10)
 
Per `evento.trascrizioneNascita.sceltaCognome` però in alcuni casi prevediamo il primo sottoinsieme, in altri il secondo e in altri ancora il terzo.
 
Questa decodifica influenza le annotazioni, sia per operazioni da compiere in caso sia modificativa sia proprio nel testo dell'annotazione.

Cordiali Saluti.
Supporto ANSC.
